detect.py

```diff
@@ -0,0 +1,374 @@
+    print("")
+    print("For more information please visit: aka.ms/ubuntu-detection")
+
+    # exit the tool 
+    sys.exit(0)
+    
+# convert dictionary data for found_commands and found_packages into .csv file format for customer to reference 
+def output_csv1(replacement_dict, data_dict, csv_filename):
+    # name csv file 
+    csv_file = csv_filename + ".csv"
+
+    # open file with write permissions 
+    with open(csv_file, 'w') as f:  
+        writer = csv.writer(f)
+
+        # iterate through dictionary containing data and add to csv file 
+        for filepath, syntax_dict in data_dict.items():
+            for syntax, line_nums in syntax_dict.items():
+                replacement = replacement_dict[syntax]
+                writer.writerow([filepath, line_nums, syntax, replacement[0], replacement[1]])
+    
+    return True
+
+
+# convert dictionary data for filepaths into .csv file format for customer to reference 
+def output_csv2(replacement_dict, data_dict, csv_filename):
+    # name csv file 
+    csv_file = csv_filename + ".csv"
+
+    # open file with write permissions 
+    with open(csv_file, 'w') as f:  
+        writer = csv.writer(f)
+
+        # iterate through dictionary containing data and add to csv file 
+        for filepath, relative_filepath in data_dict.items():
+            replacement = replacement_dict[relative_filepath]
+            writer.writerow([filepath, relative_filepath, replacement[0], replacement[1]])
+    
+    return True
+
+
+# function to add package information to the found_packages dictionary   
+def add_packages(path, packages_dict):
+    # add to the packages_dict as the value for the file path in found_packages
+    found_packages[path] = packages_dict
+
+    return True 
+
+
+# function to add command information to the found_commands dictionary   
+def add_commands(path, commands_dict):
+    # add to the command_dict as the value for the file path in found_commands
+    found_commands[path] = commands_dict
+
+    return True 
+
+
+# function to scan file for specific package names and extensions
+def package_scan(path, package):
+    # ensure path is a file by opening 
+    f = open(path)
+
+    # check to make sure f exists:
+    if not f: 
+        print("ERROR: " + path + " is not a readable file")
+        return {}
+
+    # create temp dictionary to return 
+    package_dict = {package: []}
+
+    # loop through file and check if package name is anywhere in the file 
+    # try/except for when file is unreadable 
+    try:
+        for line_num, line in enumerate(f, 1):
+            # get line and strip 
+            code = str(line.strip())
+            # check if the package is in the line
+            if package in code:
+                # if it is add it to the temp dictionary 
+                package_dict[package].append(line_num)
+    except UnicodeDecodeError:
+        # print("UnicodeDecodeError Exception")
```

Comment:
I think you forgot to uncomment this. It's good to have descriptive error/exception messages so the user can take appropriate action.